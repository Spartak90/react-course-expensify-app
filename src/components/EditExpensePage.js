import React from 'react';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
  onSubmit = () => {
    this.props.editExpense(this.props.match.params.id, this.props.expense);
    this.props.history.push('/');
  }

  render () {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    expense: state.expenses.find((expense) => {
      return expense.id === ownProps.match.params.id;
    })
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id, expense))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
