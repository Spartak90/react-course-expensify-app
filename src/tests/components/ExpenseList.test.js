import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';

const expenses = [
    { id: 1, description: 'Water bill', amount: 4500 },
    { id: 2, description: 'Rent bill', amount: 4500 },
    { id: 3, description: 'Dirt bill', amount: 4500 }
];

test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);

    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with message', () => {
    const wrapper = shallow(<ExpenseList />);

    expect(wrapper).toMatchSnapshot();
});
