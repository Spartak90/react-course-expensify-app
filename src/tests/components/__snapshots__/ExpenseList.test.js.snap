// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={4500}
    description="Water bill"
    id={1}
    key="1"
  />
  <ExpenseListItem
    amount={4500}
    description="Rent bill"
    id={2}
    key="2"
  />
  <ExpenseListItem
    amount={4500}
    description="Dirt bill"
    id={3}
    key="3"
  />
</div>
`;

exports[`should render ExpenseList with message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No Expenses on the list
  </p>
</div>
`;
